### ISSUE TITLE
Support for LSF cluster launcher

### ISSUE THREAD

**Issue by nurlanov-zh** [facebookresearch/hydra#2566]
# 🚀 Feature Request

Adding a custom plugin to support job launch on an [LSF](https://www.ibm.com/docs/en/spectrum-lsf/10.1.0) cluster.

## Motivation

Running on an LSF cluster using hydra's multirun option is problematic, since it does not allow running jobs in parallel.

## Pitch

I would like to implement a custom plugin to support LSF launcher. 

However, currently I have the following problem. I have an API from LSF cluster that allows to create jobs `submit(command, **kwargs)`, but it requires an exact `command: str` to execute as an argument.

From what I have seen, hydra does not provide the command, and instead has a function to run `self.task_function`:
```
ret = run_job(
            hydra_context=self.hydra_context,
            task_function=self.task_function,
            config=sweep_config,
            job_dir_key="hydra.sweep.dir",
            job_subdir_key="hydra.sweep.subdir",
        )
```

I am wondering if it is possible to get a command, e.g. `python example/my_app.py -experiment=example ...` , in the body of a launcher?

**Comment by Jasha10** [comment]
> I am wondering if it is possible to get a command, e.g. python example/my_app.py -experiment=example ... , in the body of a launcher?

It should be possible to write a launcher plugin that does this. Hydra doesn't have built-in support for composing a command like the example you gave -- you'd have to roll your own logic to produce the command from the launcher inputs.

**Comment by omry** [comment]
> Running on an LSF cluster using hydra's multirun option is problematic, since it does not allow running jobs in parallel.

Hydra relies on specific launchers to determine if they run jobs in parallel or not.
You can implement your own Launcher and host it in your own repo.

**Comment by mmcdermott** [comment]
While it seems like this should likely be closed as not planned in this repo, #1757 seems highly related.

**Comment by omry** [comment]
Closing as part of an automated issue cleanup sweep.

This request appears to be out of scope for this repository based on prior maintainer guidance; an external LSF launcher would be the appropriate path. If this is still relevant, or if I closed this by mistake, please comment here with updated details and we can reopen it.
